src/core/client_channel/retry_filter_legacy_call_data.h

```diff
@@ -263,6 +263,8 @@ class RetryFilter::LegacyCallData final {
     grpc_transport_stream_op_batch_payload batch_payload_;
     // For send_initial_metadata.
     grpc_metadata_batch send_initial_metadata_;
+    // For send_message.
+    absl::optional<SliceBuffer> send_message_;
```

Comment:
I don't think we need to use `absl::optional` here.  The slice buffer can always exist.  When we start a send_message op on the call attempt, we populate it from the cache.  And when the send_message op finishes on the call attempt, we can clear the slice buffer.

Reply:
done.